core-tests/jvm/src/test/scala/zio/RTSSpec.scala

```diff
@@ -1215,10 +1215,10 @@ class RTSSpec(implicit ee: ExecutionEnv) extends TestRuntime with org.specs2.mat
 
   def testRaceBothInterruptsLoser =
     unsafeRun(for {
-      s      <- Semaphore.make(0L)
+      latch  <- Promise.make[Nothing, Unit]
```

Comment:
I can't believe that degenerate behaviour was actually used in a test as a poor man's promise 😆

Reply:
I know. I want to be humble because this is the behavior of Semaphore in Java and other implementations but I remember looking at those tests being surprised that they passed at all, much less verifying expected results.